wasm/derive/src/validator/entrypoint.rs

```diff
@@ -0,0 +1,160 @@
+//! Module [`validator_entrypoint`](crate::validator_entrypoint) macro implementation
+
+use super::*;
+
+mod kw {
+    pub mod param_types {
+        syn::custom_keyword!(authority);
+        syn::custom_keyword!(transaction);
+        syn::custom_keyword!(instruction);
+        syn::custom_keyword!(query);
+        syn::custom_keyword!(expression);
+    }
+}
+
+/// Enum representing possible attributes for [`entrypoint`] macro
+enum Attr {
+    /// List of parameters
+    Params(crate::params::ParamsAttr<ParamType>),
+}
+
+impl syn::parse::Parse for Attr {
+    fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+        Ok(Attr::Params(input.parse()?))
+    }
+}
+
+/// Type of smart contract entrypoint function parameter.
+///
+/// *Type* here means not just *Rust* type but also a purpose of a parameter.
+/// So that it uses [`Authority`](ParamType::Authority) instead of `account::Id`.
+#[derive(PartialEq, Eq)]
+enum ParamType {
+    Authority,
+    Transaction,
```

Comment:
Initially there was an idea of multisig transactions checking with runtime validators. Looks like we don't like it anymore, @Erigara ?

But I think it sill may be useful for future per-user transaction validator. I.e. want to set restriction for daily spends or something like that

Reply:
So let's keep transaction validation for per-user validators as a described above?

@appetrosyan , @mversic

Reply:
This might be useful for domain and user-specific validators. Dovetails into our idea of nesting domains and having domains restrict permissions in specific ways

Reply:
Also consider the monolithic arch. In many ways Transaction validators are the only thing that we need to build up more complex logic in a static linking context.